docs/kubernetes/user-guides/snapshots.md

```diff
@@ -122,3 +122,141 @@
+PD CSI driver deployment. This may be a safer option if you are not sure about
+the deletion policy of your alpha snapshot.
+
+#### Steps to restore from an existing snapshot.
+
+  1. Confirm in
+     [console.cloud.google.com/compute/snapshots](https://console.cloud.google.com/compute/snapshots)
+     that your desired snapshots exist. Note the snapshot name, which looks like
+     `snapshot-XXXXXXXX-XXXX-XXXX-XXXX-XXXX-XXXXXXXX`, and set an env variable:
+     `export SNAPSHOT_NAME=snapshot-XXXXXXXX-XXXX-XXXX-XXXX-XXXX-XXXXXXXX` (copy
+     in your exact name).
+  1. Export your project id: `export PROJECT_ID=<your project id>`.
+  1. Create a `SnapshotVolume` resource which will be bound to a pre-provisioned
+     `SnapshotVolumeContents`. Note this is called `restored-snapshot`; this
+     name can be changed, but do it consistently across the other resources.
+     ```console
+     kubectl apply -f - <<EOF
+       apiVersion: snapshot.storage.k8s.io/v1beta1
+       kind: VolumeSnapshot
+       metadata:
+         name: restored-snapshot
+       spec:
+         volumeSnapshotClassName: csi-gce-pd-snapshot-class
+         source:
+           volumeSnapshotContentName: snapcontent-migrated
+     EOF
+     ```
+  1. Create a `SnapshotVolumeContents` pointing to your existing PD
+     snapshot.
+     ```console
+     kubectl apply -f - <<EOF
+       apiVersion: snapshot.storage.k8s.io/v1beta1
+       kind: VolumeSnapshotContent
+       metadata:
+         name: restored-snapshot-content
+       spec:
+         deletionPolicy: Retain
+         driver: pd.csi.storage.gke.io
+         source:
+           snapshotHandle: projects/$PROJECT_ID/global/snapshots/$SNAPSHOT_ID
+         volumeSnapshotRef:
+           kind: VolumeSnapshot
+           name: restored-snapshot
+           namespace: default
+     EOF
+     ```
+  1. Create a `PersistentVolumeClaim` which will pull from the
+     `VolumeSnapshot`. The `StorageClass` must match what you use to provision
+     PDs; what is below matches the example used above.
+     ```console
+     kubectl apply -f - <<EOF
+       kind: PersistentVolumeClaim
+       apiVersion: v1
+       metadata:
+         name: restored-pvc
+       spec:
+         accessModes:
+           - ReadWriteOnce
+         storageClassName: csi-gce-pd
+         resources:
+           requests:
+             storage: 6Gi
+         dataSource:
+           kind: VolumeSnapshot
+           name: restored-snapshot
+           apiGroup: snapshot.storage.k8s.io
+     EOF
+     ```
+  1. Finally, create a pod referring to the `PersistentVolumeClaim`. The PD CSI
+     driver will provision a `PersistentVolume` and populate it from the
+     snapshot. The pod from `examples/kubernetes/snapshot/restored-pod.yaml` is
+     set to use the PVC created above. After `kubectl apply`'ing the pod, run
+     `kubectl exec restored-pod -- ls -l /demo/data/` to confirm that the
+     snapshot has been restored correctly.
+
```

Comment:
You also need to uninstall the alpha CRD after you're done?

Reply:
Clarified by changing the intro above to say that this is only operating on the beta cluster.